src/core/lib/security/credentials/external/external_account_credentials.cc

```diff
@@ -229,10 +230,34 @@ void ExternalAccountCredentials::OnRetrieveSubjectTokenInternal(
   }
 }
 
+bool ValidateUrl(URI url) {
+  absl::string_view scheme(url.scheme());
+  auto string_matcher_scheme =
+      StringMatcher::Create(StringMatcher::Type::kExact, "https");
+  if (!string_matcher_scheme->Match(scheme)) {
+    return false;
+  }
+  std::vector<std::string> v = absl::StrSplit(url.authority(), ':');
+  absl::string_view host = v[0];
+  auto string_matcher_host_sts =
+      StringMatcher::Create(StringMatcher::Type::kSafeRegex,
+                            "^(([a-zA-Z0-9_]+\.sts)|(sts)|(sts\.[a-zA-Z0-9_]+)|"
```

Comment:
This regex is too restrictive, base it off this:

https:\/\/[^\.\s\/\\]+\.sts\.googleapis\.com

Reply:
Yeah, good point. While your regex doesn't work for me https://regex101.com/r/nnb760/1

Reply:
Sorry, updated my comment. Here's the ones I use for token URL (same for service account -> iamcredentials):

^[^\\.\\s\\/\\\\]+\\.sts\\.googleapis\\.com$
^sts\\.googleapis\\.com$
^sts\\.[^\\.\\s\\/\\\\]+\\.googleapis\\.com$
 ^[^\\.\\s\\/\\\\]+\\-sts\\.googleapis\\.com$

Reply:
The regexes above seem valid, but I received error `Invalid regex string specified in matcher.` Not sure why and I'm trying other formats.

Reply:
Might be an escaping issue?

try `^[^\\.\\s\\/\\\\]+\\.sts\\.googleapis\\.com$`